temporian/implementation/numpy/data/dtype_normalization.py

```diff
@@ -114,6 +114,21 @@ def normalize_features(
     `normalize_features` should match `_DTYPE_MAPPING`.
     """
 
+    def _to_bytes_array(
+        feat_array: Union[Tuple, List, np.ndarray]
+    ) -> np.ndarray:
+        """Try to convert string/object array to np.bytes or fail"""
+        try:
+            if isinstance(feat_array, np.ndarray):
+                return feat_array.astype(np.bytes_)
```

Comment:
Will this duplicate the data in memory if the original data is already bytes?

Reply:
It would, but this function is not called in that case. 
Good catch tho, because according to the docs, `astype` always copies the input array. I'll add a comment.